core/src/main/java/org/pentaho/reporting/platform/plugin/ParameterDependencyGraph.java

```diff
@@ -187,15 +196,16 @@ private boolean processDependentParameters( MasterReport report,
       final CompoundDataFactory derive = (CompoundDataFactory) cdf.derive();
       derive.initialize( factoryContext );
 
-      Boolean anyDependencyProcessed = false;
+      boolean anyDependencyProcessed = false;
       try {
         for ( final ParameterDefinitionEntry entry : reportParameter.values() ) {
           final List<String> dependentParameter = computeNormalLineage( parameterContext, entry );
           final List<String> queryDependencies = computeListParameterLineage( derive, entry, parameterValues );
 
           // No dependency information at all - No need to continue
           if ( queryDependencies == null ) {
-            return false;
+            dependencyInformationAvailable = false;
```

Comment:
Agreed, we may need to rename the variable altogether as well. Recommending changing it to allParameterDependenciesProcessed or allParametersProcessed.  That way, the semantics of it still state that either all the parameters were successfully processed or not (due to dependency query issues or exceptions).  Because we can have no dependencies and it not be an issue. The previous logic conflated No Dependencies and Dependencies Processing Issue together, which causes issues with some use cases in the report.